manager.go

```diff
@@ -0,0 +1,247 @@
+	ErrorWriter: func(ctx context.Context, w http.ResponseWriter, err error) {
+		if errgo.Cause(err) == errAuthenticationFailed {
+			// The Authenticate method has already written its response.
+			return
+		}
+		status, body := errorMapper(ctx, err)
+		httprequest.WriteJSON(w, status, body)
+	},
+}
+
+func errorMapper(ctx context.Context, err error) (int, interface{}) {
+	switch errgo.Cause(err) {
+	case ErrACLNotFound:
+		return http.StatusNotFound, &httprequest.RemoteError{
+			Message: err.Error(),
+			Code:    CodeACLNotFound,
+		}
+	case ErrBadUsername:
+		err = httprequest.Errorf(httprequest.CodeBadRequest, "%v", err)
+	}
+	return httprequest.DefaultErrorMapper(ctx, err)
+}
+
+// NewManager returns a new Manager instance that manages a
+// set of ACLs. It ensures there is at least one ACL
+// created, named "admin", which is given p.InitialAdminUsers
+// when it is first created.
+func NewManager(ctx context.Context, p Params) (*Manager, error) {
+	if err := p.Store.CreateACL(ctx, AdminACL, p.InitialAdminUsers...); err != nil {
+		return nil, errgo.Notef(err, "cannot create initial admin ACL")
+	}
+	m := &Manager{
+		p:      p,
+		router: httprouter.New(),
+	}
+	// TODO(rog) install custom NotFound handler into router?
+	httprequest.AddHandlers(m.router, reqServer.Handlers(m.newHandler))
+	return m, nil
+}
+
+// ServeHTTP implements http.Handler by serving an ACL administration
+// interface that allows clients to manipulate the ACLs. The set of
+// ACLs that can be manipulated can be changed with the Manager.NewACL
+// method.
+//
+// All the endpoints are situated underneath the RootPath prefix
+// passed to NewManager.
+func (m *Manager) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	path := req.URL.Path
+	if m.p.RootPath != "" {
+		path = strings.TrimPrefix(path, m.p.RootPath)
+		if len(path) == len(req.URL.Path) || path == "" || path[0] != '/' {
+			httprequest.WriteJSON(w, http.StatusNotFound, &httprequest.RemoteError{
+				Message: "URL path not found",
+				Code:    httprequest.CodeNotFound,
+			})
+			return
+		}
+	}
+	req.URL.Path = path
+	m.router.ServeHTTP(w, req)
+}
+
+// ACL returns the members of the given ACL.
+func (m *Manager) ACL(ctx context.Context, name string) ([]string, error) {
+	// TODO implement a cache to avoid hitting the underlying
+	// store each time.
+	return m.p.Store.Get(ctx, name)
+}
+
+// CreateACL creates an ACL with the given name. It also creates an ACL
+// _name which is the ACL that guards membership of the ACL itself. Any
+// member of _name or any member of the admin ACL may change the
+// membership of ACL name. Only members of the admin ACL may change the
+// membership of _name.
+//
+// The name itself must not start with an underscore.
+//
+// This does nothing if an ACL with that name already exists.
+func (h *Manager) CreateACL(ctx context.Context, name string, initialUsers ...string) error {
+	if isMetaName(name) {
+		return errgo.Newf("invalid ACL name %q", name)
```

Comment:
This isn't a web request, so I don't think that's appropriate here. Arguably it could return a specific error for a bad ACL name, but I can't see a use case for that right now.